### PR TITLE
fix(core): added composition dedup

### PR DESCRIPTION
**Pull request by RamilGaripov**
refs. metriport/metriport-internal#2502

### Dependencies

- Upstream: https://github.com/metriport/metriport/pull/3101

### Description
- Added `Composition` dedup to account for the large files being split into X number of chunks, which results in X number of Compositions

### Testing

- Local
  - [x] Unit tests
  - [x] Convert a few documents locally to make sure the Compositions are being deduped as expected
- Staging
  - [ ] Convert a large document on staging to make sure the Compositions are being deduped as expected

### Release Plan
- [ ] Upstream dependencies are met/released
- [ ] Merge this

### CODE REVIEW

File: packages/core/src/fhir-deduplication/__tests__/group-same-compositions.test.ts
Comment (leite08) [inline packages/core/src/fhir-deduplication/__tests__/group-same-compositions.test.ts:1]: This test is related to the `composition.ts` file, right? So it should be placed on the tests folder where that file's placed, right?

File: packages/core/src/fhir-deduplication/__tests__/group-same-compositions.test.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  compositionId2 = faker.string.uuid();
+  sourceExtension = {
+    url: DOC_ID_EXTENSION_URL,
+    valueString: createFilePath(faker.string.uuid(), faker.string.uuid(), faker.string.uuid()),
+  };
+  composition = makeComposition(undefined, { id: compositionId, extension: [sourceExtension] });
+  composition2 = makeComposition(undefined, { id: compositionId2, extension: [sourceExtension] });
+});
+
+describe("groupSameCompositions", () => {
```
Comment (leite08) [inline packages/core/src/fhir-deduplication/__tests__/group-same-compositions.test.ts:25]: We should prob also test that it won't group diff compositions, right? And that it works w/ an empty array of compositions

File: packages/core/src/fhir-deduplication/__tests__/group-same-compositions.test.ts
Comment (RamilGaripov) [reply packages/core/src/fhir-deduplication/__tests__/group-same-compositions.test.ts:1]: Ok. I'll put a TODO for all other test files to be moved as well

**Review by leite08 — APPROVED** [review]
